server/src/main/java/org/elasticsearch/script/BytesRefProducer.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script;
+
+import org.apache.lucene.util.BytesRef;
+
+public interface BytesRefProducer {
```

Comment:
Any plans to use this interface on anything other than `Version`? Otherwise maybe the instance check this is used here could simple test for `Version` instead. Or is this here so you don't need to reference `Version` from within the server module?

Reply:
The interface is there only to decouple this fix from x-pack, I'd be happy to remove it if we move Verson to server module.
Just for completeness, we could use this interface (and the same fix) for at least another case, that is [IP](https://www.elastic.co/guide/en/elasticsearch/reference/current/ip.html) field type, but the interface would still be overkill (an if/else would be more than enough probably)

Reply:
I see, thats what I guessed. No worries, I think if moving complicates things right now the interface is fine. Maybe this could benefit from a small class comment why its there then, so the next person doesn't ask again.